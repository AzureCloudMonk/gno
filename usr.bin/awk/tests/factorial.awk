# Calculate factorial of command-line argument

BEGIN {
   # Get the command-line argument
   if (ARGC < 2)   {
      print "ERROR: please provide number as command-line parameter"
      exit 1
      }
   n = ARGV[1]; ARGV[1] = "";
   # Print heading
   print "Calculating",n,"factorial"
   # Start the recursion
   nf = factorial(n);
   # All done; print the result
   print "Result:",nf;
}

# Recursive function
function factorial(n) {
   if (n <= 1)   return 1;
   else          return n*factorial(n-1);
   }
