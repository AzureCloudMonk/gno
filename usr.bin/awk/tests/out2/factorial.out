Calculating 10 factorial
Result: 3628800
